runner.py

```diff
@@ -787,10 +809,38 @@ def setup_services(self):
             if 'cmd' in service:  # must come last
                 docker_run_string.append(service['cmd'])
 
+            # before starting the container, check if the dependent containers are "ready"
+            # if not, wait for them
+            if 'depends_on' in service:
+                for dependent_container in service['depends_on']:
+                    # TODO: Make some options configurable through config.yml
+                    max_waiting_time = 20
+                    sleep_time_per_iteration = 1
+                    time_waited = 0
+                    state = ""
+                    while time_waited < max_waiting_time:
```

Comment:
I created the waiting loop for the case in which a container needs a long time to boot, and I want to wait for it until it is "ready". With the current code this is not the case, because it just checks if the container is "running" and that is the case almost immediately after it was created by Docker. To actually check if the container is ready, we need the health check.
So for now this waiting loop is most likely useless. I will remove it for this PR.
However, if the health check is implemented, something like a waiting loop is needed, I guess.

Regarding your last question: I don't think so. Therefore, I think it makes sense to throw an error immediately if a dependent container was not started/is not running.

Reply:
I would argue to keep the waiting loop in, even if it should abort in the first loop run. Maybe a comment helps that this is pre-work for having the health_check coming.

2. You are right about my sample with the alpine container. That was not representative.

But then think of a container that overwrites the entrypoint and sets CMD ["sleep", "1"]

This would exit after for example even if a terminal is attached. So our method of starting does not protect from this behaviour fully.

Thus I would argue even more to keep the loop in as it would nicely catch a directly dropping container.